Mage/MessageStream/MessageStream.cs

```diff
@@ -32,203 +39,232 @@ public class MessageStream {
-					transportClient.start();
+				if (newState == EditorPlayModeState.Playing && !transportClient.running && sessionKey != null)
+				{
+					transportClient.Start();
 				}
 			};
-#endif
-			UnityApplicationState.Instance.onAppStateChanged += (bool pauseStatus) => {
-				if (pauseStatus && transportClient.running) {
-					transportClient.stop();
+			#endif
+
+			UnityApplicationState.Instance.OnAppStateChanged += pauseStatus => {
+				if (pauseStatus && transportClient.running)
+				{
+					transportClient.Stop();
 				}
-				if (!pauseStatus && !transportClient.running && sessionKey != null) {
-					transportClient.start();
+				if (!pauseStatus && !transportClient.running && sessionKey != null)
+				{
+					transportClient.Start();
 				}
 			};
 
 			// Set the selected transport client (or the default)
-			this.SetTransport(transport);
+			SetTransport(transport);
 		}
 
 
 		//
-		public void Dispose() {
+		public void Dispose()
+		{
 			// Stop the transport client if it exists
-			if (transportClient != null) {
-				transportClient.stop();
+			if (transportClient != null)
+			{
+				transportClient.Stop();
 			}
 
-			initializeMessageList();
+			InitializeMessageList();
 			sessionKey = null;
 		}
 
 
 		// Updates URI and credentials 
-		public void SetEndpoint(string baseURL, Dictionary<string, string> headers = null) {
+		public void SetEndpoint(string baseURL, Dictionary<string, string> headers = null)
+		{
 			this.endpoint = baseURL + "/msgstream";
 			this.headers = headers;
 		}
 
 
 		// Sets up given transport client type
-		public void SetTransport(TransportType transport) {
+		public void SetTransport(TransportType transport)
+		{
 			// Stop existing transport client if any, when nulled out it will be collected
 			// by garbage collecter after existing connections have been terminated.
-			if (transportClient != null) {
-				transportClient.stop();
+			if (transportClient != null)
+			{
+				transportClient.Stop();
 				transportClient = null;
 			}
 
 			// Create new transport client instance
-			if (transport == TransportType.SHORTPOLLING) {
-				Func<string> getShortPollingEndpoint = () => {
-					return getHttpPollingEndpoint("shortpolling");
-				};
-
-				transportClient = new ShortPolling(getShortPollingEndpoint, getHttpHeaders, processMessagesString, 5000) as TransportClient;
-			} else if (transport == TransportType.LONGPOLLING) {
-				Func<string> getLongPollingEndpoint = () => {
-					return getHttpPollingEndpoint("longpolling");
-				};
-
-				transportClient = new LongPolling(getLongPollingEndpoint, getHttpHeaders, processMessagesString) as TransportClient;
-			} else {
-				throw new Exception("Invalid transport type: " + transport);
+			switch (transport)
+			{
+				case TransportType.SHORTPOLLING:
+					Func<string> getShortPollingEndpoint = () => GetHttpPollingEndpoint("shortpolling");
+					transportClient = new ShortPolling(getShortPollingEndpoint, GetHttpHeaders, ProcessMessagesString);
```

Comment:
Actually instead of making it a 1 liner i see a bug
The Func above is supposed to return a value.

The reason I dont think it can be a 1 liner is that it needs to remain a `Func<string>` that is called from within the polling system. The system is decoupled right now making the actual transport clients unaware about things like sessions etc. keeping them very lean.

Suggestions welcome though!